source/_components/camera.rpicam.markdown

```diff
@@ -0,0 +1,104 @@
+---
+layout: page
+title: "MQTT Raspberry PI camera"
+description: "Instructions on how to integrate Raspberry Pi cameras within Home Assistant."
+date: 2019-01-15 8:15
+sidebar: true
+comments: true
+sharing: true
+footer: true
+ha_category: Camera
+logo: home-assistant.png
+ha_release: 0.85
+ha_iot_class: "depends"
+---
+
+The `rpicam` camera platform allows to control a raspberry pi camera through mqtt commands.
+This is a third-party integration for homeassistant that provides further control from HA for a raspberry PI camera based on RPi-Cam-Web-Interface (`https://elinux.org/RPi-Cam-Web-Interface`).
+
+## {% linkable_title Configuration %}
+
+It can be configured like a `generic` IP camera and supports the same configuration parameters:
+
+    https://www.home-assistant.io/components/camera.generic/
+
+Set `rpicam` (instead of `generic`) as platform and follow the link above for further configuration details.
+
+Configuration example:
+
+```yaml
+      - platform: rpicam
+        name: backyard
+        still_image_url: http://192.168.1.12/cam.jpg
+        username: <USER>
```

Comment:
We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

Reply:
This new platform reuses the same configuration details as the `generic IP camera` platform and I'm going to fix the documentation to reflect the document it comes from. Unfortunately the camera name, is not optional in this case and I need to leave it in the example for consistency with the platform configuration.
I'm going to review the code as well, to ensure the constraint is enforced.